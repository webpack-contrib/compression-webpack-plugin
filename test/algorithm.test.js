import Plugin from '../src/index';

import {
  cleanErrorStack,
  createCompiler,
  compile,
  getAssetsInfo,
} from './helpers';

describe('when applied with `algorithm` option', () => {
  let compiler;

  beforeEach(() => {
    compiler = createCompiler({
      entry: {
        js: `${__dirname}/fixtures/entry.js`,
      },
    });
  });

  it('matches snapshot for `unknown` value ({String})', () => {
    expect(() => {
      new Plugin({
        minRatio: 1,
        algorithm: 'unknown',
      }).apply(compiler);
    }).toThrowErrorMatchingSnapshot();
  });

  it('matches snapshot for `gzip` value ({String})', () => {
    new Plugin({
      minRatio: 1,
      algorithm: 'gzip',
    }).apply(compiler);

    return compile(compiler).then((stats) => {
      const errors = stats.compilation.errors.map(cleanErrorStack);
      const warnings = stats.compilation.warnings.map(cleanErrorStack);

      expect(errors).toMatchSnapshot('errors');
      expect(warnings).toMatchSnapshot('warnings');
      expect(getAssetsInfo(stats.compilation.assets)).toMatchSnapshot('assets');
    });
  });

  it('matches snapshot for custom function ({Function})', () => {
    new Plugin({
      minRatio: 1,
      algorithm(input, compressionOptions, callback) {
        expect(compressionOptions).toMatchSnapshot('compressionOptions');

        return callback(null, input);
      },
    }).apply(compiler);

    return compile(compiler).then((stats) => {
      const errors = stats.compilation.errors.map(cleanErrorStack);
      const warnings = stats.compilation.warnings.map(cleanErrorStack);

      expect(errors).toMatchSnapshot('errors');
      expect(warnings).toMatchSnapshot('warnings');
      expect(getAssetsInfo(stats.compilation.assets)).toMatchSnapshot('assets');
    });
  });

  it('matches snapshot for custom function with error ({Function})', () => {
    new Plugin({
      minRatio: 1,
      algorithm(input, compressionOptions, callback) {
        expect(compressionOptions).toMatchSnapshot('compressionOptions');

        return callback('Error', input);
      },
    }).apply(compiler);

    return compile(compiler).then((stats) => {
      const errors = stats.compilation.errors.map(cleanErrorStack);
      const warnings = stats.compilation.warnings.map(cleanErrorStack);

      expect(errors).toMatchSnapshot('errors');
      expect(warnings).toMatchSnapshot('warnings');
      expect(getAssetsInfo(stats.compilation.assets)).toMatchSnapshot('assets');
    });
  });
});
