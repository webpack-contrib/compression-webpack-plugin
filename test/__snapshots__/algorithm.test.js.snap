// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when applied with \`algorithm\` option matches snapshot for \`gzip\` value ({String}): assets 1`] = `
Array [
  Array [
    "0.async.dbe59b7ea0e1a63e45d4.js",
    130,
  ],
  Array [
    "0.async.dbe59b7ea0e1a63e45d4.js.gz",
    117,
  ],
  Array [
    "5b1f36bc41ab31f5b801d48ba1d65781.png",
    78117,
  ],
  Array [
    "5b1f36bc41ab31f5b801d48ba1d65781.png.gz",
    73160,
  ],
  Array [
    "96621d3c37d96ad3bf792fcc848d912f.svg",
    672,
  ],
  Array [
    "96621d3c37d96ad3bf792fcc848d912f.svg.gz",
    393,
  ],
  Array [
    "js.b329671af0217355535e.js",
    10267,
  ],
  Array [
    "js.b329671af0217355535e.js.gz",
    2798,
  ],
]
`;

exports[`when applied with \`algorithm\` option matches snapshot for \`gzip\` value ({String}): errors 1`] = `Array []`;

exports[`when applied with \`algorithm\` option matches snapshot for \`gzip\` value ({String}): warnings 1`] = `Array []`;

exports[`when applied with \`algorithm\` option matches snapshot for \`unknown\` value ({String}) 1`] = `"Algorithm not found in zlib"`;

exports[`when applied with \`algorithm\` option matches snapshot for custom function ({Function}): assets 1`] = `
Array [
  Array [
    "0.async.dbe59b7ea0e1a63e45d4.js",
    130,
  ],
  Array [
    "0.async.dbe59b7ea0e1a63e45d4.js.gz",
    130,
  ],
  Array [
    "5b1f36bc41ab31f5b801d48ba1d65781.png",
    78117,
  ],
  Array [
    "5b1f36bc41ab31f5b801d48ba1d65781.png.gz",
    78117,
  ],
  Array [
    "96621d3c37d96ad3bf792fcc848d912f.svg",
    672,
  ],
  Array [
    "96621d3c37d96ad3bf792fcc848d912f.svg.gz",
    672,
  ],
  Array [
    "js.b329671af0217355535e.js",
    10267,
  ],
  Array [
    "js.b329671af0217355535e.js.gz",
    10267,
  ],
]
`;

exports[`when applied with \`algorithm\` option matches snapshot for custom function ({Function}): compressionOptions 1`] = `Object {}`;

exports[`when applied with \`algorithm\` option matches snapshot for custom function ({Function}): compressionOptions 2`] = `Object {}`;

exports[`when applied with \`algorithm\` option matches snapshot for custom function ({Function}): compressionOptions 3`] = `Object {}`;

exports[`when applied with \`algorithm\` option matches snapshot for custom function ({Function}): compressionOptions 4`] = `Object {}`;

exports[`when applied with \`algorithm\` option matches snapshot for custom function ({Function}): errors 1`] = `Array []`;

exports[`when applied with \`algorithm\` option matches snapshot for custom function ({Function}): warnings 1`] = `Array []`;

exports[`when applied with \`algorithm\` option matches snapshot for custom function with error ({Function}): assets 1`] = `
Array [
  Array [
    "0.async.dbe59b7ea0e1a63e45d4.js",
    130,
  ],
  Array [
    "5b1f36bc41ab31f5b801d48ba1d65781.png",
    78117,
  ],
  Array [
    "96621d3c37d96ad3bf792fcc848d912f.svg",
    672,
  ],
  Array [
    "js.b329671af0217355535e.js",
    10267,
  ],
]
`;

exports[`when applied with \`algorithm\` option matches snapshot for custom function with error ({Function}): compressionOptions 1`] = `Object {}`;

exports[`when applied with \`algorithm\` option matches snapshot for custom function with error ({Function}): compressionOptions 2`] = `Object {}`;

exports[`when applied with \`algorithm\` option matches snapshot for custom function with error ({Function}): compressionOptions 3`] = `Object {}`;

exports[`when applied with \`algorithm\` option matches snapshot for custom function with error ({Function}): compressionOptions 4`] = `Object {}`;

exports[`when applied with \`algorithm\` option matches snapshot for custom function with error ({Function}): errors 1`] = `
Array [
  "Error",
  "Error",
  "Error",
  "Error",
]
`;

exports[`when applied with \`algorithm\` option matches snapshot for custom function with error ({Function}): warnings 1`] = `Array []`;
